internal/js/modules/k6/browser/common/frame_session.go

```diff
@@ -629,6 +629,9 @@ func (fs *FrameSession) navigateFrame(frame *Frame, url, referrer string) (strin
 			err = fmt.Errorf("%q: %w", errorText, err)
 		}
 	}
+	if errorText != "" {
+		err = fmt.Errorf("navigation failed: %v", errorText)
+	}
```

Comment:
In k6 browser, when a navigation fails due to something like a DNS lookup, at the moment it continues on with the navigation and seems to assume that it succeeded. This is a problem down the line since future calls to APIs in the test script will eventually fail.

Running all the browser tests, i realised it seems that the old behaviour is valid? For example [TestBlockHostname](https://github.com/grafana/k6/blob/7129195a85e57533564e46086018e43915d27945/internal/js/modules/k6/browser/tests/network_manager_test.go#L43-L71) now fails on `page.goto`, but the test expects the `page.goto` call to succeed. What do you think the behaviour should be?

CC @inancgumus